### PR TITLE
Add GitHub actions to build plugins and run tests on PRs

### PR DESCRIPTION
**Pull request by paulyeo21**
*Description of changes:*

Add Github actions to build and run tests on push

> https://docs.github.com/en/actions/using-workflows/triggering-a-workflow

https://github.com/paulyeo21/amazon-vpc-cni-plugins/actions

[unit-test.txt](https://github.com/aws/amazon-vpc-cni-plugins/files/9740816/unit-test.txt)
[e2e-test.txt](https://github.com/aws/amazon-vpc-cni-plugins/files/9740818/e2e-test.txt)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rajal-amzn — APPROVED** [review]
Thanks for doing this!

**Review by rajal-amzn — APPROVED** [review]

File: plugins/aws-appmesh/e2eTests/e2e_test.go [hunk 1]
```diff
@@ -11,6 +11,7 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
+//go:build e2e_test || aws_appmesh
```
Comment (ofiliz) [inline plugins/aws-appmesh/e2eTests/e2e_test.go:14]: Why is this || in some, but && in others?

Comment (paulyeo21) [reply plugins/aws-appmesh/e2eTests/e2e_test.go:14]: if im not mistaken, the old annotation `+build one two` indicates or, while `+build one, two` indicates and

File: .github/workflows/integration.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
+name: integration tests
```
Comment (paulyeo21) [inline .github/workflows/integration.yaml:1]: talked to @ofiliz offline--  we can remove this

File: plugins/aws-appmesh/e2eTests/e2e_test.go (hunk 1 above)
Comment (ofiliz) [reply plugins/aws-appmesh/e2eTests/e2e_test.go:14]: Correct, but I was asking why we did it that way to begin with.
@rajal-amzn Do you need these "ecs_serviceconnect" and "aws_appmesh" tags? The e2e targets in the Makefile already run tests in that plugin's specific directory, so I don't know why we need to also have plugin-specific build tags?

Comment (rajal-amzn) [reply plugins/aws-appmesh/e2eTests/e2e_test.go:14]: No. We should be good to remove them.

**Pull request closed without merging**